### PR TITLE
feat: Vaults

### PR DESCRIPTION
**Pull request by 0xonramp**
Adds the `/vaults` page for icTBILL / icREAL

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -29,6 +29,11 @@ const nextConfig = {
       },
     ]
   },
+  webpack: (config) => {
+    config.resolve.fallback = { fs: false, net: false, tls: false }
+    config.externals.push('pino-pretty', 'lokijs', 'encoding')
+    return config
+  },
```
Comment (janndriessen) [inline next.config.js:36]: What's this for? 👀

File: src/app/vaults/page.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      amount: 10000000,
+      currency: 'USDC',
+    },
+    minimumCommitment: {
+      amount: 1000000,
+      currency: 'USDC',
+    },
+    lockupPeriod: '6 months',
+  },
+]
```
Comment (janndriessen) [inline src/app/vaults/page.tsx:41]: nice!

File: src/app/vaults/providers.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { BalanceProvider } from '@/lib/providers/Balances'
+import { ProtectionProvider } from '@/lib/providers/Protection'
+
+export function Providers({ children }: { children: React.ReactNode }) {
+  return (
+    <BalanceProvider>
+      <ProtectionProvider>{children}</ProtectionProvider>
+    </BalanceProvider>
+  )
```
Comment (janndriessen) [inline src/app/vaults/providers.tsx:9]: Probably won't need balance provider here but can be left for now

File: src/lib/hooks/useLedgerStatus.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 export function useLedgerStatus() {
-  if (!window) return { isRunningInLedgerLive: false }
+  if (typeof window === 'undefined') return { isRunningInLedgerLive: false }
```
Comment (janndriessen) [inline src/lib/hooks/useLedgerStatus.ts:2]: Oh 👀 this fixes that annoying error/warning? nice!

**Review by janndriessen — COMMENTED** [review]
Looks good for now. Only the shadow of the button might not be correct yet.

File: next.config.js (hunk 1 above)
Comment (0xonramp) [reply next.config.js:36]: Fixes the console warnings while running in dev mode: https://github.com/WalletConnect/walletconnect-monorepo/issues/1908#issuecomment-1487801131

(Matches our implementation in `interfaces`: https://github.com/IndexCoop/interfaces/blob/main/apps/auctions/next.config.js#L16)

File: src/app/globals.css [hunk 5]
```diff
@@ -6,7 +6,7 @@
     'Fira Mono', 'Droid Sans Mono', 'Courier New', monospace;
 
   --foreground-rgb: 0, 0, 0;
-  --background-start-rgb: 214, 219, 220;
+  --background-start-rgb: 255, 255, 255;
```
Comment (0xonramp) [inline src/app/globals.css:9]: @janndriessen Ok to make this change? (e.g. the dark mode isn't used)

File: src/lib/hooks/useLedgerStatus.ts (hunk 4 above)
Comment (0xonramp) [reply src/lib/hooks/useLedgerStatus.ts:2]: @janndriessen maybe?! I saw it surface via a build failure: https://vercel.com/index-coop/index-app/9uhEv1buM5Dzob62mP7vihoYhGEB#L88

Is that where you saw the error/how did it surface for you?

Comment (janndriessen) [reply src/lib/hooks/useLedgerStatus.ts:2]: I sometimes saw this in the console. Not 100 percent sure it's the same but good fix nevertheless.

**Pull request merged**